src/state_spaces/real_vector_space/RealVectorSpace.cpp

```diff
@@ -7,17 +7,17 @@
 #include "RealVectorSpaceConfig.h"
 #include "xArm6.h"
 
-base::RealVectorSpace::RealVectorSpace(int num_dimensions_) : StateSpace(num_dimensions_)
+base::RealVectorSpace::RealVectorSpace(size_t num_dimensions_) : StateSpace(num_dimensions_)
 {
 	srand((unsigned int) time(0));
-	setStateSpaceType(StateSpaceType::RealVectorSpace);
+	setStateSpaceType(base::StateSpaceType::RealVectorSpace);
 }
 
-base::RealVectorSpace::RealVectorSpace(int num_dimensions_, const std::shared_ptr<robots::AbstractRobot> robot_, 
+base::RealVectorSpace::RealVectorSpace(size_t num_dimensions_, const std::shared_ptr<robots::AbstractRobot> robot_, 
 	const std::shared_ptr<env::Environment> env_) : StateSpace(num_dimensions_, robot_, env_)	
 {
 	srand((unsigned int) time(0));
```

Comment:
This might me unnecessary with <random> and/or Eigen.

Reply:
It is not clear to me what do you mean...